### DISCUSSION TITLE
Is CodeQL will cover OWASP top 10 vulnerabilities?

### DISCUSSION THREAD

**Discussion by PoleMadhu123** [community/community#67948] [Code Security] [answered]
### Select Topic Area

Question

### Body

Want to know is the CodeQL will cover vulnerabilities against to OWASP Top 10 vulnerabilities.
These are the top 10 OWASP vulnerabilities:
1. Broken Access control
2. Cryptographic Failures
3. Injection
4. Insecure Design
5. Security Misconfiguration
6. Vulnerable and Outdated components
7. Identification and Authentication Failures
8. Software and Data Integrity Failures
9. Security Logging and Monitoring Failures
10. Server-Side Request Forgery (SSRF) Failures

**Comment by aibaars** [comment, marked as answer]
CodeQL should have queries for all of the owasp categories. The depth of analysis may vary between languages. See also https://codeql.github.com/codeql-query-help/codeql-cwe-coverage/
